src/S3/S3Client.php

```diff
@@ -239,80 +239,107 @@ class S3Client extends AwsClient implements S3ClientInterface
     /** @var array */
     private static $mandatoryAttributes = ['Bucket', 'Key'];
 
+    /** @var array */
+    private static $checksumOptionEnum = [
+        'when_supported' => true,
+        'when_required' => true
+    ];
+
     public static function getArguments()
     {
         $args = parent::getArguments();
         $args['retries']['fn'] = [__CLASS__, '_applyRetryConfig'];
         $args['api_provider']['fn'] = [__CLASS__, '_applyApiProvider'];
 
-        return $args + [
-            'bucket_endpoint' => [
-                'type'    => 'config',
-                'valid'   => ['bool'],
-                'doc'     => 'Set to true to send requests to a hardcoded '
-                    . 'bucket endpoint rather than create an endpoint as a '
-                    . 'result of injecting the bucket into the URL. This '
-                    . 'option is useful for interacting with CNAME endpoints.',
-            ],
-            'use_arn_region' => [
-                'type'    => 'config',
-                'valid'   => [
-                    'bool',
-                    Configuration::class,
-                    CacheInterface::class,
-                    'callable'
+        return
+            [
+                'request_checksum_calculation' => [
+                    'type' => 'config',
+                    'valid' => ['string'],
+                    'doc' => 'Set to true to disable the usage of'
```

Comment:
This parameter's doc still needs to be updated.

Reply:
another silly mistake